temporal-sdk/src/main/java/io/temporal/internal/replay/ReplayWorkflowTaskHandler.java

```diff
@@ -101,20 +101,119 @@ public ReplayWorkflowTaskHandler(
     String workflowType = workflowTask.getWorkflowType().getName();
     Scope metricsScope =
         options.getMetricsScope().tagged(ImmutableMap.of(MetricsTag.WORKFLOW_TYPE, workflowType));
-    try {
-      if (workflowTask.hasQuery()) {
-        // Legacy query codepath
+    if (workflowTask.hasQuery()) {
+      // Legacy query codepath
+      try {
         return handleQueryOnlyWorkflowTask(workflowTask.toBuilder(), metricsScope);
-      } else {
+      } catch (Throwable e) {
+        return createQueryResult(workflowTask, Optional.empty(), e);
+      }
+    } else {
+      try {
         return handleWorkflowTaskWithEmbeddedQuery(workflowTask.toBuilder(), metricsScope);
+      } catch (Throwable e) {
+        metricsScope.counter(MetricsType.WORKFLOW_TASK_EXECUTION_FAILURE_COUNTER).inc(1);
+        return failureToWFTResult(workflowTask, e);
       }
+    }
+  }
+
+  private Result handleWorkflowTaskWithEmbeddedQuery(
+      PollWorkflowTaskQueueResponse.Builder workflowTask, Scope metricsScope) throws Throwable {
+    AtomicBoolean createdNew = new AtomicBoolean();
+    WorkflowExecution execution = workflowTask.getWorkflowExecution();
+    WorkflowRunTaskHandler workflowRunTaskHandler = null;
+    boolean useCache = stickyTaskQueueName != null;
+
+    try {
+      workflowRunTaskHandler =
+          getOrCreateWorkflowExecutor(useCache, workflowTask, metricsScope, createdNew);
+      WorkflowTaskResult result = workflowRunTaskHandler.handleWorkflowTask(workflowTask);
+      if (result.isFinalCommand()) {
+        cache.invalidate(execution, metricsScope, "FinalCommand", null);
+      } else if (useCache && createdNew.get()) {
+        cache.addToCache(execution, workflowRunTaskHandler);
+      }
+      return createCompletedWFTRequest(
+          workflowTask.getWorkflowType().getName(), workflowTask, result);
     } catch (Throwable e) {
```

Comment:
Can any of the things in here throw themselves?

Reply:
Well, resetStickyTaskQueue actually can, good thinking. I will wrap it into a catch block that ignores it and logs instead.

Reply:
Actually no, it doesn't throw, we use the Future non-blocking version of the call.